src/components/HTMLEngineProvider/htmlEngineUtils.js

```diff
@@ -44,4 +44,21 @@ function isInsideComment(tnode) {
     return false;
 }
 
-export {computeEmbeddedMaxWidth, isInsideComment, isCommentTag};
+/**
+ * Check if there is an ancestor node with the name 'h1'.
+ * Finding a node with the name 'h1' flags that we are rendering inside an h1 element.
+ * @param {TNode} tnode
+ * @returns {Boolean}
+ */
+function isInsideH1(tnode) {
+    let currentNode = tnode;
```

Comment:
```suggestion
    let currentNode = tnode.parent;
```

We should skip the current node otherwise this may return a false positive e.g. when the tnode is actually the h1. But it's not insider another h1

Reply:
@yakupafsin This is not resolved yet